sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -199,6 +209,19 @@ private[sql] case class FileSourceScanExec(
         options = relation.options,
         hadoopConf = relation.sparkSession.sessionState.newHadoopConfWithOptions(relation.options))
 
+      (file: PartitionedFile) => {
+        val iter = func(file)
+        // Only for test purpose.
+        // Once the vectorized Parquet reader is initialized in the above method, we can read its
+        // variable numRowGroups.
+        if (fileFormat != null) {
```

Comment:
These looks ugly, can we create a named accumulator here, then pass that into VectorizedParquetRecordReader and update it there?

We could add an private api in AccumulatorContext to lookup by name.